src/backend/c-lisp.py

```diff
@@ -0,0 +1,288 @@
+                else:
+                    return self.gen_expr(stmt)
+            else:
+                return self.gen_expr(stmt)
+        except Exception as e:
+            print(f"Error in statement: {stmt}")
+            raise e
+
+    def is_for_stmt(self, stmt):
+        return stmt[0] == "for"
+
+    def gen_for_stmt(self, stmt):
+        if len(stmt) < 3:
+            raise CodegenError(f"Bad for statement: {stmt}")
+
+        cond_sym, loop_lbl, cont_lbl, break_lbl = [
+            random_label(prefix)
+            for prefix in (
+                "tmp_clisp_cond",
+                "tmp_clisp_loop",
+                "tmp_clisp_cont",
+                "tmp_clisp_break",
+            )
+        ]
+        init_expr_instr = self.gen_expr(stmt[1][0])
+        cond_expr_instr = self.gen_expr(stmt[1][1], res_sym=cond_sym)
+        iter_expr_instr = self.gen_expr(stmt[1][2])
+        loop_stmt_instr = self.gen_stmt(stmt[2:])
+
+        return [
+            *init_expr_instr,
+            ["label", loop_lbl],
+            *cond_expr_instr,
+            ["br", cond_sym, cont_lbl, break_lbl],
+            ["label", cont_lbl],
+            *loop_stmt_instr,
+            *iter_expr_instr,
+            ["jmp", loop_lbl],
+            ["label", break_lbl],
+        ]
+
+    def is_if_stmt(self, stmt):
+        return stmt[0] == "if"
+
+    def gen_if_stmt(self, stmt):
+        cond_sym, true_lbl, false_lbl, out_lbl = [
+            random_label(prefix)
+            for prefix in (
+                "tmp_clisp",
+                "lbl_true_clisp",
+                "lbl_false_clisp",
+                "lbl_out_clisp",
+            )
+        ]
+
+        cond_instr_list = self.gen_expr(stmt[1], res_sym=cond_sym)
+        true_instr_list = self.gen_stmt(stmt[2])
+        if len(stmt) == 3:
+            false_instr_list = []
+        elif len(stmt) == 4:
+            false_instr_list = self.gen_stmt(stmt[3])
+        else:
+            raise CodegenError(f"Bad if statement: {stmt}")
+
+        return [
+            *cond_instr_list,
+            ["br", cond_sym, true_lbl, false_lbl],
+            ["label", true_lbl],
+            *true_instr_list,
+            ["jmp", out_lbl],
+            ["label", false_lbl],
+            *false_instr_list,
+            ["label", out_lbl],
+        ]
+
+    def is_decl_stmt(self, stmt):
+        return stmt[0] == "declare"
+
+    def gen_decl_stmt(self, stmt):
+        if not (len(stmt) == 2 and len(stmt[1]) == 2):
+            raise CodegenError(f"bad declare statement: {stmt}")
+
+        name, typ = stmt[1]
+        if name in self.symbol_types:
+            raise CodegenError(f"Re-declaration of variable {name}")
+        self.symbol_types[name] = typ
+        return []
+
+    def is_ret_stmt(self, stmt):
+        return stmt[0] == "ret"
+
+    def gen_ret_stmt(self, stmt):
+        if len(stmt) == 1:
+            return [["ret"]]
+        elif len(stmt) == 2:
+            res_sym = random_label("tmp_clisp")
```

Comment:
Keep a global constant at the top like CLISP_PREFIX = "tmp_clisp" and reuse this variable everywhere

Reply:
Okay, forget class. Maybe just a wrapper

Reply:
I've done this, using a wrapper. What do you think of it?

Reply:
Done